src/Symfony/Bundle/WebProfilerBundle/EventListener/ProfilerLinkLogListener.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\EventListener;
+
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+use Symfony\Component\HttpKernel\Event\ResponseEvent;
+use Symfony\Component\HttpKernel\KernelEvents;
+use Symfony\Component\Routing\Generator\UrlGeneratorInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @author Jérémy Romey jeremyFreeAgent <jeremy@free-agent.fr>
+ */
+class ProfilerLinkLogListener implements EventSubscriberInterface
+{
+    public function __construct(
+        protected ?LoggerInterface $logger = null,
+        private ?UrlGeneratorInterface $urlGenerator = null,
+    ) {
+    }
+
+    public function onKernelResponse(ResponseEvent $event): void
+    {
+        if (null === $this->logger) {
+            return;
+        }
+        if (null === $this->urlGenerator) {
+            return;
+        }
+
+        $response = $event->getResponse();
+        $request = $event->getRequest();
+
+        if (!$event->isMainRequest()) {
+            return;
+        }
+
+        if (false === $response->headers->has('X-Debug-Token')) {
+            return;
+        }
+
+        $this->logger->debug(\sprintf('See profiler at %s', $this->urlGenerator->generate('_profiler', ['token' => $response->headers->get('X-Debug-Token')], UrlGeneratorInterface::ABSOLUTE_URL)));
```

Comment:
The logger should do the [interpolation](https://www.php-fig.org/psr/psr-3/#12-message). Does it work like this?
```suggestion
        $this->logger->debug('See profiler at {profiler_url}', ['profiler_url' => $this->urlGenerator->generate('_profiler', ['token' => $response->headers->get('X-Debug-Token')], UrlGeneratorInterface::ABSOLUTE_URL)]);
```

Reply:
why not use directly `X-Debug-Token-Link` header?